source/common/grpc/async_client_impl.cc

```diff
@@ -139,6 +139,8 @@ void AsyncStreamImpl::onHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_s
 }
 
 void AsyncStreamImpl::onData(Buffer::Instance& data, bool end_stream) {
+  stream_->streamInfo().addBytesReceived(data.length());
```

Comment:
Q: should this be done in the HTTP async client rather than gRPC?

Reply:
This is a good question. I initially added this to grpc async client because (1) wanted to limited to envoy gRPC which is initial motivation of this PR (2) avoid the collision with [existing addBytes](https://github.com/envoyproxy/envoy/blob/87876cfe8fe411a4ec5f491b94587196ad355382/source/common/http/async_client_impl.cc#L327) in http async client ( which seems to be unused code though)

However, adding bytes metering in http async client should work as well. Basically, we need to 
1.  Add `addBytesReceived` in [http encodeData](https://github.com/envoyproxy/envoy/blob/87876cfe8fe411a4ec5f491b94587196ad355382/source/common/http/async_client_impl.cc#L138) and add `addBytesSent` in  [http sendData](https://github.com/envoyproxy/envoy/blob/87876cfe8fe411a4ec5f491b94587196ad355382/source/common/http/async_client_impl.cc#L174) 
2. Remove existing addBytes in [http onData](https://github.com/envoyproxy/envoy/blob/87876cfe8fe411a4ec5f491b94587196ad355382/source/common/http/async_client_impl.cc#L327). Because this will be covered by encodeData above.  
For http: http_encodeData -> http_onData.  For Envoy gRPC: http_encodeData -> grpc_onData

I can see the benefits of adding them to http async client. It is lower layer implementation that covers both envoy gRPC and http, though the byte metering in http async stream is not required for the moment.